src/OpenTelemetry.Instrumentation.Process/ProcessMetrics.cs

```diff
@@ -58,4 +68,86 @@ static ProcessMetrics()
     public ProcessMetrics(ProcessInstrumentationOptions options)
     {
     }
+
+    private class InstrumentsValues
+    {
+        private const int TicksPerSecond = 10 * 7;
+        private readonly Diagnostics.Process currentProcess = Diagnostics.Process.GetCurrentProcess();
+
+        private double? memoryUsage;
+        private double? virtualMemoryUsage;
+        private double? cpuTime;
+        private double? cpuUtlization;
+
+        private DateTime lastCollectionTimestamp = DateTime.Now;
+        private double lastCollectionCpuTime;
+
+        public InstrumentsValues(Func<(double MemoryUsage, double VirtualMemoryUsage, double CpuTime, double CpuUtilization)> readValues)
+        {
+            this.memoryUsage = null;
+            this.virtualMemoryUsage = null;
+            this.cpuTime = null;
+            this.cpuUtlization = null;
+
+            this.currentProcess.Refresh();
+            this.UpdateValues = readValues;
+        }
+
+        private Func<(double MemoryUsage, double VirtualMemoryUsage, double CpuTime, double CpuUtilization)> UpdateValues { get; set; }
+
+        public double GetMemoryUsage()
+        {
+            if (!this.memoryUsage.HasValue)
+            {
+                (this.memoryUsage, this.virtualMemoryUsage, this.cpuTime, this.cpuUtlization) = this.UpdateValues();
+            }
+
+            var value = this.currentProcess.WorkingSet64;
+            this.memoryUsage = null;
+            return value;
+        }
+
+        public double GetVirtualMemoryUsage()
+        {
+            if (!this.virtualMemoryUsage.HasValue)
+            {
+                (this.memoryUsage, this.virtualMemoryUsage, this.cpuTime, this.cpuUtlization) = this.UpdateValues();
+            }
+
+            var value = this.currentProcess.VirtualMemorySize64;
+            this.virtualMemoryUsage = null;
+            return value;
+        }
+
+        public double GetCpuTime()
+        {
+            if (!this.cpuTime.HasValue)
+            {
+                (this.memoryUsage, this.virtualMemoryUsage, this.cpuTime, this.cpuUtlization) = this.UpdateValues();
+            }
+
+            var value = this.currentProcess.TotalProcessorTime.Ticks * TicksPerSecond;
+            this.cpuTime = null;
+            return value;
+        }
+
+        public double GetCpuUtilization()
+        {
+            if (!this.cpuUtlization.HasValue)
+            {
+                (this.memoryUsage, this.virtualMemoryUsage, this.cpuTime, this.cpuUtlization) = this.UpdateValues();
+            }
+
+            var elapsedTime = (DateTime.Now - this.lastCollectionTimestamp).Ticks * TicksPerSecond;
```

Comment:
To convert ticks to seconds, you need to divide the ticks by the TicksPerSecond.

Reply:
right, this is embarrassing, thanks for pointing it out.